cluster/lock.go

```diff
@@ -172,6 +177,29 @@ func (l Lock) VerifySignatures() error {
 		return errors.Wrap(err, "verify lock signature aggregate")
 	}
 
+	if len(l.NodeSignatures) != 0 {
```

Comment:
please move this to `Lock.VerifySignatures` remove implicit verification from marshalling, only do explicit verification via `VerifySignatures`

Reply:
not sure I got what you mean, this check is already in `Lock.VerifySignature`